src/java/org/apache/cassandra/config/AccordSpec.java

```diff
@@ -38,6 +38,20 @@ public class AccordSpec
 
     public volatile boolean enable_journal_compaction = true;
 
+    /**
+     * Enables the virtual Accord debug keyspace with tables
+     * that expose internal state to aid the developers working
+     * on Accord implementation.
+     * <p/>
+     * These tables can and will change and/or go away at any point,
+     * including in a minor release, and are NOT to be relied on for anything.
+     * <p/>
+     * Only enable this keyspace if you are working on Accord and
+     * need to debug an issue with Accord implementation, or if an Accord
+     * developer asked you to.
+     */
+    public boolean enable_virtual_debug_keyspace = false;
```

Comment:
this documentation is not enough for saying the tables are not exposed to the public API.  

See `Re: [EXTERNAL] Re: [DISCUSS] Adding experimental vtables and rules around them` for the dev thread about this topic.

To unblock you I am fine withs something like `enable_unsupported_virtual_tables` as the name tells you they are `unsupported`, but I am a hard -1 on merging Accord to trunk with this topic not dealt with.  As a project we need to define a way to have experimental (aka unsupported) tables, and until that is done Accord can not merge to trunk as it is the first to define such tables...

Reply:
These are neither experimental nor unsupported. They are purely internal. I don't know how much clearer one can be.

Reply:
To to be as explicit as possible: these tables are intended for human consumption only. To my knowledge it is not project policy that virtual tables must be consumable by applications, but even if you take that they are then flagging them off resolves any concern. This is all we mean by API-unstable. Really we mean "NOT AN API, ONLY AN HMI". 

We will never provide an API for viewing _implementation details_ while I have a say, as it would prevent reasonable development. But, it is helpful to expose these details to operators for human interpretation of the cluster state _only_.

I am aware of no project policy that would be in conflict with merge. So, if you are opposed to merging CEP-15 once this has merged, please decide what you require to proceed and take it to the list for discussion.

Reply:
>  But, it is helpful to expose these details to operators for human interpretation of the cluster state only.

If the assumption is to expose this for operators, then operators will build tools as manual work is a massive waste of operator time.  So breaking the table becomes a massive burden for operators, who you are saying are a target audience.  

If the audience is tests, then flagging on for tests and off for prod is fine for me, we do this with system properties today.

> To my knowledge it is not project policy that virtual tables must be consumable by applications,

at no point have I said "consumable by applications", I said "public api".  We have had a long stance that every table and every JMX is part of our public API and we can not break.  I have also said there is a desire and willingness to define breakable tables, but we must do it in a way that actually documents that tables are breakable, which is one of the main reasons the ML has focused on how do we expose this detail to users / operators.  If someone sees `debug` do they think this can change at a moments notice, or that this is useful to debug the issues you are facing?  The focus on the ML has been the word "experimental" as the focus was on tables not 100% ready for production, but this patch wants a different definition, one I don't think anyone is opposed to... my concern is articulating this to people, I want to avoid people being blind sided by a breaking change.

Tl;DR - if this is for tests or people working on accord patches and won't be enabled for prod, I am mostly fine, but still think a better config name would help give this message.

If the audience is operations this then means enabled on prod, then I don't think the config name or documentation tells operators what rules they should expect around the API changing.

Reply:
I was previously an operator, and manually using introspection views on e.g. SQL Server was a perfectly normal activity, and I did not use scripts to access this information. By "Consumable by applications" I meant "API" (the A means application)

I'm obviously not _opposed_ to folk using scripts, just so long as they accept the possibility that their scripts may need tweaking when they go to use them, because we don't promise not to break them. If the scripts are just labour saving devices that the operator understands, this shouldn't be overly onerous, or destroy the value proposition.

The alternative in my view is either _not providing these views_, or providing ones with massively reduced information. One option might be to have restricted versions of each view, and a config flag that enriches the tables with more information, and makes more tables available. But I would prefer to explore that closer to release of 5.1, as in the meantime many of these tables will change anyway.

I'm agnostic regarding naming of any flag, and also perfectly supportive of migrating this to some endorsed "unstable API" mechanism as and when it becomes available.

Reply:
> The alternative in my view is either not providing these views, or providing ones with massively reduced information

I am fine with the progress log tables being off by default and treating it as testing code.  I am 100% fine saying we will flesh out an operator focused table closer to merge or release, and that isn't required scope of this patch.